test/models/product_test_test.rb

```diff
@@ -126,7 +126,7 @@ def compare_records(patient1, patient2)
       de2 = patient2.dataElements.fetch(x)
       de1.attributes.each do |k, v|
         assert_nil de2.attributes[k], 'random repeatability error: dataElements different, non-nil match' if v.nil?
-        assert_equal v, de2.attributes[k], 'random repeatability error: dataElements different' unless k == '_id'
+        assert_equal v, de2.attributes[k], 'random repeatability error: dataElements different' unless %w[_id id].include?(k)
```

Comment:
Could you document here why we're getting both `id` and `_id` attributes here for clarity?